src/adapter/src/coord.rs

```diff
@@ -2998,25 +2998,33 @@ impl Coordinator {
                 .system_config()
                 .coord_slow_message_warn_threshold();
 
+            const MESSAGE_BATCH: usize = 64;
+            let mut messages = Vec::with_capacity(MESSAGE_BATCH);
+            let mut cmd_messages = Vec::with_capacity(MESSAGE_BATCH);
+
+            let message_batch = self.metrics
+                .message_batch
+                .with_label_values(&[]);
+
             loop {
                 // Before adding a branch to this select loop, please ensure that the branch is
                 // cancellation safe and add a comment explaining why. You can refer here for more
                 // info: https://docs.rs/tokio/latest/tokio/macro.select.html#cancellation-safety
-                let msg = select! {
+                select! {
                     // Order matters here. Some correctness properties rely on us processing
                     // internal commands before processing external commands.
```

Comment:
This comment, the fact that some correctness properties rely on the order that we process commands, gives me pause on adding batching. What I'm thinking of is:

Previously
```
1. two commands in cmd_rx, process the first which emits an event to internal_cmd_rx
2. process command from internal_cmd_rx
3. process second command from cmd_rx
```

Now
```
1. two commands in cmd_rx, process the first which emits an event to internal_cmd_rx, then process the second
2. process command from internal_cmd_rx
```

Now the event sent to `internal_cmd_rx` no longer get processed before the second command in `cmd_rx`. I _think_ this is okay but we should definitely do a nightly run before merging